control-plane/agents/core/src/core/registry.rs

```diff
@@ -313,4 +313,12 @@ impl Registry {
             None => Ok(()),
         }
     }
+
+    /// Returns whether or not the node with the given ID is cordoned.
+    pub fn node_cordoned(&self, node_id: &NodeId) -> bool {
+        self.specs
+            .get_cordoned_nodes()
+            .into_iter()
+            .any(|node_spec| node_spec.id() == node_id)
```

Comment:
Would it be simpler to get just the node that has `node_id` and check if it's cordoned?

Reply:
Good suggestion. Done